charon/src/transform/mod.rs

```diff
@@ -127,6 +128,8 @@ pub static ULLBC_PASSES: &[Pass] = &[
     // # Micro-pass: remove the drops of locals whose type is `Never` (`!`). This
     // is in preparation of the next transformation.
     UnstructuredBody(&remove_drop_never::Transform),
+    // Monomorphize the functions and types.
+    UnstructuredBody(&monomorphize::Transform),
```

Comment:
Your pass doesn't depend on details of llbc vs ullbc, could you make it work for both so that we can run it after other cleanup passes?